networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/client_nsx.py

```diff
@@ -43,17 +45,59 @@ def __call__(cls, *args, **kwargs):
 
 
 class RetryPolicy(object):
-
-    def _create_sentry_fingerprint(path: str):
-        # check if uuid is part of path -> replace with {}
+    BASE = "/policy/api/v1"
+    BASE_INFRA = f"{BASE}/infra"
+    RULES = [
+        # Regex Order matters
+        # Match security_rule before security_policy
+        # Match ports before segments
+        ('security_rule', re.compile(f"^{BASE_INFRA}/domains/default/security-policies/.*/rules.*")),
+        ('security_policy', re.compile(f"^{BASE_INFRA}/domains/default/security-policies.*")),
+        ('group', re.compile(f"^{BASE_INFRA}/domains/default/groups.*")),
+        ('port', re.compile(f"^{BASE_INFRA}/segments/.*/ports.*")),
+        ('segments', re.compile(f"^{BASE_INFRA}/segments.*")),
+        ('realized_state', re.compile(f'^{BASE_INFRA}/realized-state/status.*')),
+        ('services', re.compile(f"^{BASE_INFRA}/services.*")),
+        ('search', re.compile(f"^{BASE}/search.*")),
+        ('transport_zone', re.compile(f"^{BASE_INFRA}/sites/default/enforcement-points/default/transport-zones.*")),
+        ('qos_profile', re.compile(f"^{BASE_INFRA}/default/qos-profiles.*")),
+    ]
+
+    @staticmethod
+    def _create_sentry_fingerprint(path: str, placeholder: str = "{}") -> str:
+        # check if uuid is part of path -> replace with placeholder
         for sub in path.split("/"):
             try:
                 uuid.UUID(sub)
-                path =  path.replace(sub, placeholder)
+                path = path.replace(sub, placeholder)
             except ValueError:
                 pass
         return path
 
+    @staticmethod
+    def _get_resource_type(path: str) -> str:
+        for name, rule in RetryPolicy.RULES:
+            match = rule.match(path)
+            if match:
+                return name
+        return "unknown"
+
+    @staticmethod
+    def _update_metric(metric, **kwargs):
+        path = kwargs.get("path", '')
+        new_path = RetryPolicy._create_sentry_fingerprint(path=path, placeholder="<uuid>")
+        resource = RetryPolicy._get_resource_type(path)
+
+        if "histogram:nsxv3_agent_api_calls" == str(metric):
```

Comment:
You could use `if str(RetryPolicy.API_CALLS) == str(metrics):` or probably even without the `str()`